### ISSUE TITLE
dials.find_spots on stills deeply broken if shoebox=false

### ISSUE THREAD

**Issue by graeme-winter** [dials/dials#186]
Running tests just now, it just ... sits there

**Comment by graeme-winter** [comment]
This is the problem (found if nproc=1)

```
DIALS 1.dev.568-g86dc717
The following parameters have been modified:

output {
  shoeboxes = False
}
input {
  datablock = datablock.json
}

Setting spotfinder.filter.min_spot_size=3 Configuring spot finder from input parameters
--------------------------------------------------------------------------------
Finding strong spots in imageset 0
--------------------------------------------------------------------------------


Finding spots in image 1 to 750...
Setting chunksize=25
Extracting strong spots from images
 Using multiprocessing with 1 parallel job(s)

Found 2 strong pixels on image 1
Traceback (most recent call last):
  File "/dls/science/users/gw56/svn/cctbx/build/../modules/dials/command_line/find_spots.py", line 196, in <module>
    halraiser(e)
  File "/dls/science/users/gw56/svn/cctbx/build/../modules/dials/command_line/find_spots.py", line 194, in <module>
    script.run()
  File "/dls/science/users/gw56/svn/cctbx/build/../modules/dials/command_line/find_spots.py", line 143, in run
    datablocks[0], params)
  File "/dls/science/users/gw56/svn/cctbx/modules/dials/array_family/flex.py", line 175, in from_observations
    return find_spots(datablock)
  File "/dls/science/users/gw56/svn/cctbx/modules/dials/algorithms/spot_finding/finder.py", line 775, in __call__
    table, hot_mask = self._find_spots_in_imageset(imageset)
  File "/dls/science/users/gw56/svn/cctbx/modules/dials/algorithms/spot_finding/finder.py", line 865, in _find_spots_in_imageset
    r, h = extract_spots(imageset[j0:j1])
  File "/dls/science/users/gw56/svn/cctbx/modules/dials/algorithms/spot_finding/finder.py", line 477, in __call__
    return self._find_spots_2d_no_shoeboxes(imageset)
  File "/dls/science/users/gw56/svn/cctbx/modules/dials/algorithms/spot_finding/finder.py", line 696, in _find_spots_2d_no_shoeboxes
    reflections.extend(function(task)[0])
  File "/dls/science/users/gw56/svn/cctbx/modules/dials/algorithms/spot_finding/finder.py", line 238, in __call__
    reflections, _ = converter(self.imageset, pixel_labeller)
  File "/dls/science/users/gw56/svn/cctbx/modules/dials/algorithms/spot_finding/finder.py", line 412, in __call__
    imageset, pixel_labeller)
  File "/dls/science/users/gw56/svn/cctbx/modules/dials/algorithms/spot_finding/finder.py", line 316, in __call__
    self.write_hot_pixel_mask)
Boost.Python.ArgumentError: Please report this error to dials-support@lists.sourceforge.net: Python argument types in
    PixelListShoeboxCreator.__init__(PixelListShoeboxCreator, PixelListLabeller, int, int, bool, int, int, FilterRunner) did not match C++ signature:
    __init__(_object*, dials::model::PixelListLabeller pixel, unsigned long panel=0, unsigned long zstart=0, bool twod=False, unsigned long min_pixels=1, unsigned long max_pixels=20, bool find_hot_pixels=False)

```

**Comment by graeme-winter** [comment]
Wondering about closing this, but the issue remains that the parallel part of the code never returned so leaving this open @jmp1985 - we should really fix that. Tested core part of code, though regression test needed.

**Comment by Anthchirp** [comment]
patched for 1.3.1

**Comment by graeme-winter** [comment]
Obliged
